protocols/kad/src/behaviour.rs

```diff
@@ -930,21 +927,17 @@ where
     /// when a new peer is inserted in the routing table.
     /// This parameter is used to call [`Behaviour::bootstrap`] periodically and automatically
     /// to ensure a healthy routing table.
-    pub fn bootstrap(&mut self) -> Result<QueryId, NoKnownPeers> {
+    pub fn bootstrap(&mut self) -> QueryId {
+        self.bootstrap_status.on_started();
         let local_key = *self.kbuckets.local_key();
         let info = QueryInfo::Bootstrap {
             peer: *local_key.preimage(),
             remaining: None,
             step: ProgressStep::first(),
         };
         let peers = self.kbuckets.closest_keys(&local_key).collect::<Vec<_>>();
-        if peers.is_empty() {
-            Err(NoKnownPeers())
-        } else {
-            self.bootstrap_status.on_started();
-            let inner = QueryInner::new(info);
-            Ok(self.queries.add_iter_closest(local_key, peers, inner))
-        }
+        let inner = QueryInner::new(info);
+        self.queries.add_iter_closest(local_key, peers, inner)
     }
```

Comment:
Yes this was my first approach and it does fix the problem but I though you were not completely happy about it.

If you're ok with it, I'm perfectly happy to merge this change since it does fix the problem and it seems several people are experiencing it. However, I do think it is important that users can be notified when an automatic / periodic bootstrap fail at startup, because since this feature, there is no way to react on it.

Reply:
I've implemented this solution. If you're ok with it I think we should merge it before the next release.

Reply:
I have made a new suggestion, but if you don't like it, we can go ahead with this solution.

Reply:
No it is perfectly fine. Thanks alot. I'm resolving all the conversations since it seems everyone is ok with the latest solution :wink: